runelite-client/src/main/java/net/runelite/client/plugins/emojis/EmojiPlugin.java

```diff
@@ -166,6 +222,134 @@ public void onOverheadTextChanged(final OverheadTextChanged event)
 		event.getActor().setOverheadText(updatedMessage);
 	}
 
+	@Subscribe
+	public void onMenuEntryAdded(MenuEntryAdded event)
+	{
+		int childId = WidgetInfo.TO_CHILD(event.getActionParam1());
+		if (!config.showEmitEmojiEntry() || childId != WidgetInfo.RESIZABLE_VIEWPORT_ALL_CHAT_TAB.getChildId())
```

Comment:
Why is this only comparing child IDs? And why is it comparing only to `RESIZABLE_VIEWPORT_ALL_CHAT_TAB`? What happened to checking `FIXED_VIEWPORT_ALL_CHAT_TAB`?

Reply:
The child Id of the All tab on fixed and resize is the same, so I didn't catch that. I'll check both verbosely.

Do we need to check the Group Id of the `MenuEntryAdded` too?

Reply:
@Nightfirecat how do we check the exact id of the widget instead of the child id? Can we do something like this? 

```
int widgetId = event.getActionParam1();
if (!config.showEmitEmojiEntry()
	|| widgetId != WidgetInfo.RESIZABLE_VIEWPORT_ALL_CHAT_TAB.getId()
	|| widgetId != WidgetInfo.FIXED_VIEWPORT_ALL_CHAT_TAB.getId())
{
	return;
}
```

I've tried this but I'm not sure if it's correct or not, it's not working but it's possible the `WidgetInfo.RESIZABLE_VIEWPORT_ALL_CHAT_TAB` and `FIXED_VIEWPORT_ALL_CHAT_TAB` are just wrong (since I've added them). Not sure which id in the widget inspector to use

Reply:
That code looks correct to me. The widget inspector will show the group and child IDs in the tree on the left, and should display the `ID` as a row when selecting the widget.

On second thought, it will likely be easier to use the MenuManager class to manage adding these custom menu entries to these widgets.

Reply:
I've refactored to use the MenuManager helper. Let me know how it looks.